src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -1468,16 +1424,16 @@ private IndexWriter createWriter() throws IOException {
             mergePolicy = new ElasticsearchMergePolicy(mergePolicy);
             config.setMergePolicy(mergePolicy);
             config.setSimilarity(similarityService.similarity());
-            config.setRAMBufferSizeMB(indexingBufferSize.mbFrac());
-            config.setMaxThreadStates(indexConcurrency);
-            config.setCodec(codecService.codec(codecName));
+            config.setRAMBufferSizeMB(this.config.getIndexingBufferSize().mbFrac());
+            config.setMaxThreadStates(this.config.getIndexConcurrency());
+            config.setCodec(codecService.codec(this.config.getCodecName()));
```

Comment:
Can we name this `writerConfig`? It's weird to shadow the class `config` since it looks like the same thing (though it isn't)